src/script/standard.cpp

```diff
@@ -88,21 +88,55 @@ static constexpr bool IsSmallInteger(opcodetype opcode)
     return opcode >= OP_1 && opcode <= OP_16;
 }
 
-static bool MatchMultisig(const CScript& script, unsigned int& required, std::vector<valtype>& pubkeys)
+static constexpr bool IsPushdataOp(opcodetype opcode)
+{
+    return opcode > OP_FALSE && opcode <= OP_PUSHDATA4;
+}
+
+/** Valid number of keys in a multisig are [1,20]. */
+static constexpr bool IsValidMultisigKeyCount(unsigned int n_keys)
+{
+    return n_keys > 0 && n_keys <= MAX_PUBKEYS_PER_MULTISIG;
+}
+
+static bool GetMultisigKeyCount(opcodetype opcode, valtype data, int& count)
+{
+    if (IsSmallInteger(opcode)) {
+        count = CScript::DecodeOP_N(opcode);
+        return IsValidMultisigKeyCount(count);
+    }
+
+    if (IsPushdataOp(opcode)) {
+        try {
+            count = CScriptNum(data, true).getint();
```

Comment:
The caller should (and does) only regard the value of `count` if the getter returns `true`.
I could make sure that on error we always return before writing to `count` but it would make the logic even more convoluted than it already is in order to support an usage that should not exist in the first place?